### PR TITLE
feat(engine-util): reorg interceptor

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Supersedes #8110.

Create a stream wrapper that can simulate reorgs for block n based on the contents of block n + 1.

### CODE REVIEW

File: crates/engine/util/src/reorg.rs [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+                            // Enter the reorg state.
+                            // The current payload will be immediately forwarded by being in front
+                            // of the queue. Then we attempt to reorg the current head by generating
+                            // a payload that attaches to the head's parent and is based on the
+                            // non-conflicting transactions (txs from block `n + 1` that are valid
+                            // at block `n` according to consensus checks) from the current payload
+                            // as well as the corresponding forkchoice state. We will rely on CL to
+                            // reorg us back to canonical chain.
+                            // TODO: This is an expensive blocking operation, ideally it's spawned
+                            // as a task so that the stream could yield the control back.
```
Comment (mattsse) [inline crates/engine/util/src/reorg.rs:155]: should be okay for debugging

File: crates/engine/util/src/reorg.rs [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+    // Configure environments
+    let mut cfg = CfgEnvWithHandlerCfg::new(Default::default(), Default::default());
+    let mut block_env = BlockEnv::default();
+    evm_config.fill_cfg_and_block_env(
+        &mut cfg,
+        &mut block_env,
+        chain_spec,
+        &reorg_target.header,
+        U256::MAX, // TODO:
```
Comment (mattsse) [inline crates/engine/util/src/reorg.rs:269]: this is okay because no longer relevant for beacon consensus

**Review by mattsse — APPROVED** [review]
I'm realizing that we could improve a lot of building ergonomics...

this is not integrated into the EthNodeLauncher but we can do that separately

File: crates/engine/util/src/reorg.rs (hunk 2 above)
Comment (rkrasiuk) [inline crates/engine/util/src/reorg.rs:269]:
```suggestion
        U256::MAX,
```

**Pull request merged**